### ISSUE TITLE
Preserve background process from initContainers to runner

### ISSUE THREAD

**Issue by robooo** [grafana/k6-operator#394]
### Feature Description

My current deployment requires to do vpn connections which are done by executing a process in the background e.g. `./scripts/vpn &`.  If I don't execute it in background then k6 pod is waiting forever. I'm not able to change the connection approach/logic.

I hoped initContainers can help me, like I can setup the interface and other commands but can't keep alive the background process for k6 execution.

### Suggested Solution (optional)

I'm not sure if allowing shareProcessNamespace: true would solve this?
Based on this https://stackoverflow.com/questions/65576193/how-to-keep-a-process-running-throughout-pod-lifecycle

If there is some easier way please let me know.

**Comment by yorugac** [comment]
Hi @robooo, thanks for opening the issue!

This is an interesting case. AFAIS, `shareProcessNamespace` is more meant to help with debugging. So IMHO, your case sounds like it requires another regular container instead of `initContainer` ([docs](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/#differences-from-regular-containers)). 

This never came up before and there is an argument to be made that k6 container in the runner pod shouldn't have any other containers in order to avoid skewing results of the test.

> I'm not able to change the connection approach/logic.

Out of curiosity, why is it not possible? I.e. "normally", Kubernetes access can be set up so that one won't need to run custom VPN scripts inside.

**Comment by robooo** [comment]
Thanks @yorugac! I understand, but I don't want to create some new "proxy pod" with installed binaries and running vpn and then route k6 via it, since it can be a bottleneck. Or are there some recommended deployment examples in case of using vpn + k6? 
 
I just really want to execute the binaries before k6 run to be able to access the backend and route the traffic via it. The init container was the first approach which I've tried and yesterday I found another way with [xk6-exec](https://github.com/grafana/xk6-exec) which allowed me to execute the vpn binary right from the test case (in the k6 container) after few tweaks I'm able to run it in the background too. But wouldn't it be easier to allow `command` for k6 runner or something similar?

**Comment by yorugac** [comment]
This topic actually reminded of [multi-cluster problem](#240) and related projects, like [this one](https://kilo.squat.ai/).

**Comment by robooo** [comment]
@yorugac exactly! I'm trying to test/reach webserver from outside of Kubernetes cluster that is reachable only via VPN. It is no openvpn or something opensourced but more internal. The next step will be execute it from a different cluster in cloud not from local private server but I think the approach needs to be the same.

**Comment by yorugac** [comment]
Hello!
Re-reading this issue in context of a [future PodTemplate](https://github.com/grafana/k6-operator/issues/815): ATM, I'm inclined to think that the command of k6 container will remain constructed by k6-operator and not exposed to a user in PodTemplate. It's too tricky to implement a good UX for such modifications, TBH. Command is a string array: how k6-operator is supposed to merge an array and remain functional _and_ readable in general case, while following PodTemplate schema? And there is a workaround with `xk6-exec` too, at least for OSS usage.

It'd be curious to hear if there is any other use case for the modification of command, other than a background process like VPN.

**Comment by robooo** [comment]
Yes I believe you it's hard to do something UX friendly here :/ 
Back then, that was the only use case for me.
